File: Assets/Bundles/Shared/Resources/TS/Util/RandomUtil.lua
-- Compiled with unity-ts v2.1.0-75
local RandomUtil
do
	RandomUtil = setmetatable({}, {
		__tostring = function()
			return "RandomUtil"
		end,
	})
	RandomUtil.__index = RandomUtil
	function RandomUtil.new(...)
		local self = setmetatable({}, RandomUtil)
		return self:constructor(...) or self
	end
	function RandomUtil:constructor()
	end
	function RandomUtil:FromArray(array)
		local index = math.random(0, #array - 1)
		return array[index + 1]
	end
end
return {
	RandomUtil = RandomUtil,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Shared/Util/RandomUtil.ts
export class RandomUtil {
	/**
	 * Fetch a random element from an array.
	 * @param array An array.
	 * @returns A random element from `array`.
	 */
	public static FromArray<T>(array: T[]): T {
		const index = math.random(0, array.size() - 1);
		return array[index];
	}
}
